src/hrflow_connectors/connectors/lever/warehouse.py

```diff
@@ -0,0 +1,396 @@
+        description="Authorization code for obtaining access token",
+        field_type=FieldType.Auth,
+    )
+    limit: int = Field(
+        100,
+        description="Number of jobs to fetch per request (max: 100)",
+        field_type=FieldType.QueryParam,
+    )
+
+
+class WriteProfileParameters(ParametersModel):
+    client_id: str = Field(
+        ...,
+        description="Client ID for authenticating with Lever API",
+        field_type=FieldType.Auth,
+    )
+    client_secret: str = Field(
+        ...,
+        description="Client secret for authenticating with Lever API",
+        field_type=FieldType.Auth,
+    )
+    authorization_code: str = Field(
+        ...,
+        description="Authorization code for obtaining access token",
+        field_type=FieldType.Auth,
+    )
+    limit: int = Field(
+        100,
+        description="Number of jobs to fetch per request (max: 100)",
+        field_type=FieldType.QueryParam,
+    )
+    perform_as: str = Field(
+        ...,
+        description="User ID on behalf of whom the create action should be performed",
+        field_type=FieldType.QueryParam,
+    )
+    parse: bool = Field(
+        False,
+        description="If true, parse resume for autofilling",
+        field_type=FieldType.QueryParam,
+    )
+    perform_as_posting_owner: bool = Field(
+        False,
+        description="If true, set Opportunity owner to posting owner",
+        field_type=FieldType.QueryParam,
+    )
+
+
+class ReadProfilesParameters(ParametersModel):
+    client_id: str = Field(
+        ...,
+        description="Client ID for authenticating with Lever API",
+        field_type=FieldType.Auth,
+    )
+    client_secret: str = Field(
+        ...,
+        description="Client secret for authenticating with Lever API",
+        field_type=FieldType.Auth,
+    )
+    authorization_code: str = Field(
+        ...,
+        description="Authorization code for obtaining access token",
+        field_type=FieldType.Auth,
+    )
+    limit: int = Field(
+        100,
+        description="Number of jobs to fetch per request (max: 100)",
+        field_type=FieldType.QueryParam,
+    )
+
+
+def get_or_refresh_tokens(
+    client_id, client_secret, grant_type, authorization_code=None, refresh_token=None
+):
+    url = "https://sandbox-lever.auth0.com/oauth/token"
+    redirect_uri = "https://marketplace-partners.hrflow.ai/partner/lever/login"
+    if authorization_code:
+        request_data = {
+            "client_id": client_id,
+            "client_secret": client_secret,
+            "grant_type": grant_type,
+            "code": authorization_code,
+            "redirect_uri": redirect_uri,
+        }
+    elif refresh_token:
+        request_data = {
+            "client_id": client_id,
+            "client_secret": client_secret,
+            "grant_type": grant_type,
+            "refresh_token": refresh_token,
+        }
+    else:
+        raise Exception("Invalid grant type")
```

Comment:
En fait ici tu pourrais mieux communiquer ton intention en changeant les arguments de la fonction. 
Parce que si à la fois `refresh_token` et `authorization_code` sont `None` tu dis `Invalid grant type`. Ca peut surprendre. 

Alors que la valeur de `grant_type` donnée en argument par l'utilisateur peut être valide. Je pense que ce que tu veux faire est plutot du genre 
```python
def get_or_refresh(client_id, client_secret, authorization_code=None, refresh_token=None): 
    if authorization_code is None and refresh is None:
        raise TypeError("Either of authorization_code or refresh_token should not be None") 

    # Ce check est necessaire aussi parce que dans ton implementation actuelle rien empeche un utilisateur
    # de mettre à la fois authorization_code et refresh_token 
    if authorization_code and refresh_token: 
        raise TypeError("Only one of authorization_code or refresh_token should be not None") 

    if authorization_code: 
          # Ici tu mettre directement le bon grant_type pas besoin de demander à l'utilisateur 
```

Reply:
en fait, en relisant le code suite à tes remarques ici, je vois que ce n'est pas utile de gerer ce cas. En fait, le grant type est alimenté toujours en statique dans le code donc sera toujours l'un des valeurs
![Capture d’écran du 2023-11-13 16-47-26](https://github.com/Riminder/hrflow-connectors/assets/135601200/7ad9d68f-f57e-493b-8fd9-9938590083b6)
![Capture d’écran du 2023-11-13 16-46-55](https://github.com/Riminder/hrflow-connectors/assets/135601200/3bda4ccf-0208-40dc-a6c1-234ee14db017)

Reply:
En fait un `argument` ou une variable c'est comme son nom l'indique quelque chose qui est susceptible de varier. 

Ici la définir `grant_type` comme argument c'est à la fois peu utile et éventuellement source d'erreur. 

La raison c'est parce que `grant_type` n'est pas une vrai variable dans ton code. Parce que si `authorization_code` n'est pas `None` ca **doit** être "authorization_code` et si `refresh_token` n'est pas `None` ca **doit** être `refresh_token`. 

C'est pas une variable parce que ca valeur depend directement d'autres variables. 

Donc c'est inutile parce que tu peux obtenir la bonne valeur à partir d'un autre argument. Et c'est source d'erreur parce que quelqu'un peut se tromper et faire `get_or_refresh(refresh_token="xxxx", grant_type="authorization_token")` 

Mnt ta réponse de dire dans le code c'est utilisé de manière statique c'est pas pertinent. 
Parce que la le raisonnement est au niveau de la fonction. 
Une fois que tu définis une fonction avec des arguments X ou Y ca doit marcher pour tout l'ensemble des possiblités que tu as défini. 

Si toi en tant qu'utilisateur tu sais que tu as pas besoin de l'argument X parce que ca ne varie pas dans tes utilisations actuelles alors tu fais une fonction sans X. 

Mais tu fais pas une fonction avec X qui est cassé dans le cas général en disant oui mais dans mon code je fais que des utilisations valides. 

Donc pars plutôt sur une signature comme ca 
```
import typing as t 

def get_or_refresh_token(
      auth_domain: str, 
      client_id: str,
      client_secret: str, 
      flow: t.Literal["authorization_code", "refresh_token"],
      code: str
)
     ....
```
La tu as les avantages suivant: 
- Le code n'est plus cassé. Avant rien n'empechait de mettre à la fois `authorization_code` et `refresh_token` . Ni de mettre `authorization_code` et une valeur incorrecte dans `grant_type` 
- Tu as enlever une variable 
- Tu dis clairement que tu géres deux flow seulement `authorization_code` ou `refresh_token` et tu forces l'utilisateur à forcement mettre soit l'un soit l'autre 



Ici tu as une dependence forte entre `authorization_code` `refresh_token`

Reply:
très bien pensé merci. j'ai implementé cela

Reply:
t.Literal apparemment n'est pas fonctionnel en python 3.7

Reply:
A solution is to use typing_extensions instead of typing to success importing literal in python3.7

Reply:
ça a resolu le porblem des core tests, c'est reussi maintenant